### PR TITLE
Prepend JMSDiExtraBundle configuration making FosRestBundle/NelmioApiDocBundle optional

### LINKED ISSUE #686: Getting errors because of API code bundle needs
Hey,

I have `"sonata-project/user-bundle": "^2.2"` in my composer.json. 
I do not plan to use any type of API component. When I do `composer update`  I get this:

```
[Doctrine\Common\Annotations\AnnotationException]
  [Semantical Error] The annotation "@Nelmio\ApiDocBundle\Annotation\ApiDoc" in method Sonata\UserBundle\Controller\Api\GroupController::getGroupsAction() does not exist, or could not be auto-loaded.


Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the post-update-cmd event terminated with an exception



  [RuntimeException]
  An error occurred when executing the "'cache:clear --no-warmup'" command:

    [Doctrine\Common\Annotations\AnnotationException]
    [Semantical Error] The annotation "@Nelmio\ApiDocBundle\Annotation\ApiDoc" in method Sonata\UserBundle\Controller\Api\GroupController::getGroupsAction() does not exist, or could not be auto-loaded.

  .
```

Do I need to install & configure FOSRestBundle and NelmioApiDocBundle? Is this something that you are forcing users to do???

### PR DESCRIPTION
**Pull request by gremo**
I am targetting this branch, because this PR make possible to use JMSDiExtra bundle without `FosRestBundle` and `NelmioApiDocBundle`.

Closes #686

## Changelog

```markdown
### Fixed
- make it possible to use JMSDiExtra bundle without FosRestBundle and NelmioApiDocBundle
```

### CODE REVIEW

File: DependencyInjection/SonataUserExtension.php [hunk 1]
```diff
@@ -356,6 +357,32 @@ public function configureMenu(array $config, ContainerBuilder $container)
...
+     * {@inheritdoc}
+     */
+    public function prepend(ContainerBuilder $container)
+    {
+        $bundles = $container->getParameter('kernel.bundles');
+
+        // Blacklist all API controllers when using JMSDiExtraBundle without NelmioApiDocBundle or FOSRestBundle. API
+        // controllers use annotations and would be warmed, throwing an exception.
+        if (isset($bundles['JMSDiExtraBundle']) &&
+            (!isset($bundles['NelmioApiDocBundle']) || !isset($bundles['FOSRestBundle']))) {
```
Comment (greg0ire) [inline DependencyInjection/SonataUserExtension.php]: I think this is equivalent to `!isset($bundles['NelmioApiDocBundle'], $bundles['FOSRestBundle'])`

Comment (gremo) [inline DependencyInjection/SonataUserExtension.php]: I'll check and report back, never used more than one argument.

Comment (core23) [inline DependencyInjection/SonataUserExtension.php]: Please update this PR @gremo

Comment (gremo) [inline DependencyInjection/SonataUserExtension.php]: @core23 done!

**Comment by OskarStark** [comment]
@gremo could you please finalize this PR?

**Comment by gremo** [comment]
@OskarStark I rebased this PR as a starting point. How should I finalize this? I however still think that the problem is described https://github.com/schmittjoh/JMSDiExtraBundle/issues/269 but no one cares...

**Comment by gremo** [comment]
So @OskarStark this is a workaround because no one seems to care the [PR](https://github.com/schmittjoh/JMSDiExtraBundle/issues/269) that I opened to JMSDiExtraBundle. How we should proceed?

**Comment by gremo** [comment]
So @greg0ire I think we can close this thanks to https://github.com/schmittjoh/JMSDiExtraBundle/pull/270?

**Pull request closed without merging**